bgpd/bgp_io.c

```diff
@@ -148,12 +149,21 @@ static int bgp_process_writes(struct thread *thread)
 		fatal = true;
 	}
 
+	/* If suppress fib pending is enabled, route is advertised to peers when
+	 * the status is received from the FIB. The delay is added
+	 * to update group packet generate which will allow more routes to be
+	 * sent in the update message
+	 */
+	if (BGP_SUPPRESS_FIB_ENABLED(peer->bgp) &&
+	    PEER_ROUTE_ADV_DELAY(peer))
+		val = BGP_DEFAULT_UPDATE_ADVERTISEMENT_TIME;
```

Comment:
Protocol specific details do not belong in I/O code. Since this idiom appears in the FSM code as well please make it a new wrapper function around the BGP_TIMER_ON call on line 166, and use it here and in the FSM code. That wrapper function probably belongs next to `bgp_generate_updgrp_packets`.